subsys/bluetooth/host/smp.h

```diff
@@ -38,7 +38,11 @@ struct bt_smp_hdr {
 #define BT_SMP_OOB_NOT_PRESENT			0x00
 #define BT_SMP_OOB_PRESENT			0x01
 
+#if defined(CONFIG_BT_SMP)
```

Comment:
Is this really necessary? Do you get build warnings without this?

Reply:
Unfortunately yes, as CONFIG_BT_SMP_MIN_ENC_KEY_SIZE isn't visible unless CONFIG_BT_SMP=y, so if smp.h is included while CONFIG_BT_SMP=n then it give a build error. 

This is a general issue with the Kconfig system if using `if <config>` to add dependency to multiple other config options, and this has been the way to deal with that in many cases. 

If you have any other ideas on how to deal with this, I'd very much like to hear them, as this isn't very clean in my opinion.

Reply:
Only case I’d expect errors is if code both includes and *uses* the define. I wouldn’t expect that to happen  if SMP is not enabled. Hmm, or is it some IS_ENABLED() block that’s always visible to the compiler that causes the error? Where is it?

Reply:
No, just referring the Kconfig option in a header file, regardless of it is used, gives this error.

Reply:
Could you quote the error you get?

Reply:
@jhedberg https://buildkite.com/zephyr/zephyr/builds/25716

Reply:
Hmm, that might actually be a different case. Let me do some additional testing tomorrow.

Reply:
Thanks! That’s a bit odd though - why is smp.c getting compiled if CONFIG_SMP is not set? Don’t get me wrong, I’m not unposed to the ifdef if it can’t be avoided, but I’d like to thoroughly understand the case here.

Reply:
Looking at CMakeLists.txt smp.c is only built if CONFIG_SMP is set, so the error you referenced doesn’t really make sense (unless there’s something obvious I don’t understand). Looking at the timestamps of the build log, is it possible it’s for a version where you had an additional `depends on` for the Kconfig symbol? Ie did you actually verify that it’s still an issue for this latest version where no such dependency exists? Again, sorry if I’m misusing something obvious, but it just doesn’t make sense to me 🙂

Reply:
Indeed, the issue above was not caused by the issue I was referencing.

Reply:
@jhedberg 

Found another example where this was necessary (I don't think it's necessary in this scenario actually, and have reverted it). But if you look in https://github.com/zephyrproject-rtos/zephyr/blob/master/include/bluetooth/mesh/cdb.h you can see a similar "solution" as the `#define` is used in the same file: 

```
#if defined(CONFIG_BT_MESH_CDB)
#define NODE_COUNT    CONFIG_BT_MESH_CDB_NODE_COUNT
#define SUBNET_COUNT  CONFIG_BT_MESH_CDB_SUBNET_COUNT
#define APP_KEY_COUNT CONFIG_BT_MESH_CDB_APP_KEY_COUNT
#else
#define NODE_COUNT    0
#define SUBNET_COUNT  0
#define APP_KEY_COUNT 0
#endif
```

Without that, including cdb.h without CONFIG_BT_MESH_CDB would give a build error.

Reply:
For mesh I think those defines are used for array sizes that are I side structs  in the same header file), so the end result is invalid C code if the defines don’t evaluate to integers. So it’s a different issue there.

Reply:
Indeed :)